### DISCUSSION TITLE
crash in RAND_priv_bytes but runing continue

### DISCUSSION THREAD

**Discussion by MuYuLinZhong** [openssl/openssl#25866] [Q&A]
I use `RAND_priv_bytes` under `openssl 1.1.1m` to get bytes，and I got core dump in this function as many issues in this project.

![image](https://github.com/user-attachments/assets/05eb9dbc-a9d6-477c-8cf7-246d28de7d42)

And it is clear to see the lock is empty. But I feel confused:
1. I can not find where the lock release, I don't clean up the lock in my program. But actually it only crash in exit process.
2. I find that it will continue runing after core dump, and there is no error return to my program. So my program can continue to exit down.

could somebody can point out the reason, thank you

**Comment by MuYuLinZhong** [comment]
By the way, I use `RAND_priv_bytes` many time in my program, but it always core dump at last. It is also odd...
![image](https://github.com/user-attachments/assets/91e1f841-416a-461d-81cf-6bd653917f83)

**Comment by nhorman** [comment]
1.1.1m is long out of support, you would have to recreate this issue under a newer version of openssl (3.0 or later)

Though for what its worth, its pretty clear that a lock wasn't allocated for your context when calling RAND_priv_bytes.  If it only crashes sometimes, that sounds like it might be a race in library cleanup from another thread

**Reply by MuYuLinZhong** [reply]
Yes, I reviewed problems like mine under issus board, but it seems doesn't have a common answer for this question.

And I am confused. When some thread have exit(-1), and will sleep for a while in atexit function, openssl will cleanup by itself? I don't take cleanup function in my program.
